### PR TITLE
Upgraded greenlight core lightning version to 24.02.1.

### PR DESCRIPTION
**Pull request by Randy808**
Rebased and made this branch into a draft pull request so it's ready to merge when we've upgraded the cln version on our servers

- [x] Requires core lightning v24.02.1 to be pushed to cloud storage as a greenlight artifact. Otherwise check-self will fail with `make: *** [Makefile:148: cln-versions/lightningd-v24.02.1gl1.tar.bz2] Error 8`

### CODE REVIEW

File: libs/gl-testing/gltesting/utils.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ def is_compat(self, nv: NodeVersion) -> bool:
             "v22.11": ["v22.11gl1"],
             "v23.05": ["v23.05gl1"],
             "v23.08": ["v23.08gl1"],
+            "v24.02.1": ["v23.02.1gl1"],
```
Comment (nepet) [inline libs/gl-testing/gltesting/utils.py:21]: Typo? `v24.02.1` != `v23.02.1`

**Comment by nepet** [comment]
Also, an update of the signer version is needed to upgrade the node to v24.02. Can be found in `signer/mod.rs` : `VERSION`. This is called on `maybe_upgrade`.

File: libs/gl-testing/gltesting/utils.py (hunk 1 above)
Comment (Randy808) [reply libs/gl-testing/gltesting/utils.py:21]: Yeah, a typo. Good catch 👍

**Comment by cdecker** [comment]
Rebased on top of `master`

**Comment by nepet** [comment]
`libs/gl-testing/node.py` class `NodeProcess` is missing the `--developer` flag on versions >= `24.02`. Adding the following snippets fixes this for the time being until pyln-testing supports multiple versions.

```python
#libs/gl-testing/gltesting/utils.py

def version_to_int(nv: NodeVersion):
        # Remove the initial 'v'
        version_string = nv.name.lstrip('v')

        # Split the version string into parts by '.' and strip any non-numeric suffix
        parts = version_string.split('.')
        numeric_parts = []
        
        for part in parts:
            numeric_part = ''.join(filter(str.isdigit, part))
            numeric_parts.append(numeric_part.zfill(2))  # Zero-fill to ensure 2 digits per part
        
        # Ensure there are exactly 4 parts, padding with '00' if necessary
        while len(numeric_parts) < 3:
            numeric_parts.append('00')
        
        # Join the parts into a single string
        parsed_version = ''.join(numeric_parts[:3])  # Take only the first 4 parts

        return int(parsed_version)
```

```python
#libs/gl-testing/gltesting/utils.py

class NodeVersion(TailableProcess):
...

   self.cmd_line = [
            str(self.executable),
            f'--lightning-dir={self.directory}',
            f'--network={network}',
            '--log-level=debug',
            '--bitcoin-rpcuser=rpcuser',
            '--bitcoin-rpcpassword=rpcpass',
            f'--bitcoin-rpcconnect=127.0.0.1:{self.bitcoind.rpcport}',
            "--disable-plugin=commando",
            '--rescan=1',
            "--log-timestamps=false",
            "--cltv-final=6",
            f"--addr=127.0.0.1:{self.p2p_port}",
            # Stock `cln-grpc` disabled in favor of `gl-plugin`
            '--disable-plugin=cln-grpc',
            f'--subdaemon=hsmd:{signerproxy_path}',
            f'--important-plugin={plugin_path}',
            '--dev-bitcoind-poll=5',
            '--dev-fast-gossip',
            '--offline',
            '--experimental-anchors',
        ]

        if version_to_int(self.version) >= 240200:
            self.cmd_line.append('--developer');

...
```

This reduces the checks that fail to

`sendcustommsg` changed on pyln-client (no keyword args):
 - `test_node.py::test_custommsg`
 - `test_lsps.py::test_lsps_list_protocol`
 
some signer errors: `- ValueError: Some signer reported an error: [msg: "resolver error:`
 - libs/gl-testing/tests/test_gl_node.py::test_configure_close_to_addr
 - libs/gl-testing/tests/test_node.py::test_node_invoice_amountless
 - libs/gl-testing/tests/test_node.py::test_lsp_jit_fee
 - libs/gl-testing/tests/test_node.py::test_vls_crash_repro
 - libs/gl-testing/tests/test_node.py::test_sendpay_signer 
 - libs/gl-testing/tests/test_node.py::test_node_listpays_preimage


Other tests are good!

**Review by cdecker — APPROVED** [review]

**Pull request merged**